File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int x = Integer.parseInt(br.readLine());

        Arrays.sort(arr);

        int l = 0;
        int r = n - 1;
        int cnt = 0;
        while (l < r) {
            int mid = arr[l] + arr[r];
            if (mid == x) {
                cnt++;
                l++;
                r--;
            } else if (mid < x) {
                l++;
            } else if (mid > x) {
                r--;
            }
        }

        System.out.println(cnt);
    }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;
    
    vector<int> v;
    
    for (int i = 0; i < n; i++) {
        int a;
        cin >> a;
        v.push_back(a);
    }
    
    sort(v.begin(), v.end());
    
    int x;
    cin >> x;
    
    int cnt = 0;
    int l = 0;
    int r = n - 1;
    while (l < r) {
        int mid = v[l] + v[r];
        if (mid == x) {
            cnt++;
            l++;
            r--;
        } else if (mid < x) {
            l++;
        } else if (mid > x) {
            r--;
        }
    }
    
    cout << cnt;
    return 0;
}